x/stakeibc/keeper/msg_server_redeem_stake_test.go

```diff
@@ -0,0 +1,260 @@
+	stakeibc "github.com/Stride-Labs/stride/x/stakeibc/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	_ "github.com/stretchr/testify/suite"
+)
+
+type RedeemStakeTestCase struct {
+	user         Account
+	module       Account
+	initialState State
+	validMsg     stakeibc.MsgRedeemStake
+}
+
+func (suite *KeeperTestSuite) SetupRedeemStake() RedeemStakeTestCase {
+	redeemAmount := int64(1_000_000)
+	initialDepositAmount := int64(1_000_000)
+	user := Account{
+		acc:           suite.TestAccs[0],
+		atomBalance:   sdk.NewInt64Coin("ibc/uatom", 10_000_000),
+		stAtomBalance: sdk.NewInt64Coin("stuatom", 10_000_000),
+	}
+	suite.FundAccount(user.acc, user.atomBalance)
+	suite.FundAccount(user.acc, user.stAtomBalance)
+
+	module := Account{
+		acc:           suite.App.AccountKeeper.GetModuleAddress(stakeibc.ModuleName),
+		atomBalance:   sdk.NewInt64Coin("ibc/uatom", 10_000_000),
+		stAtomBalance: sdk.NewInt64Coin("stuatom", 10_000_000),
+	}
+	suite.FundModuleAccount(stakeibc.ModuleName, module.atomBalance)
+	suite.FundModuleAccount(stakeibc.ModuleName, module.stAtomBalance)
+
+	// TODO define the host zone with stakedBal and validators with staked amounts
+	hostZone := stakeibc.HostZone{
+		ChainId:        "GAIA",
+		HostDenom:      "uatom",
+		Bech32Prefix:   "cosmos",
+		RedemptionRate: sdk.NewDec(1.0),
+		StakedBal:      1234567890,
+	}
+
+	epochTrackerDay := stakeibc.EpochTracker{
+		EpochIdentifier: epochtypes.DAY_EPOCH,
+		EpochNumber:     1,
+	}
+
+	epochUnbondingRecord := recordtypes.EpochUnbondingRecord{
+		Id:                   1,
+		UnbondingEpochNumber: 1,
+		HostZoneUnbondings:   make(map[string]*recordtypes.HostZoneUnbonding),
+	}
+
+	epochUnbondingRecord.HostZoneUnbondings["GAIA"] = &recordtypes.HostZoneUnbonding{
+		Amount:     uint64(0),
+		Denom:      "uatom",
+		HostZoneId: "GAIA",
+		Status:     recordtypes.HostZoneUnbonding_BONDED,
+	}
+
+	suite.App.StakeibcKeeper.SetHostZone(suite.Ctx, hostZone)
+	suite.App.StakeibcKeeper.SetEpochTracker(suite.Ctx, epochTrackerDay)
+	suite.App.RecordsKeeper.AppendEpochUnbondingRecord(suite.Ctx, epochUnbondingRecord)
+
+	return RedeemStakeTestCase{
+		user:   user,
+		module: module,
+		initialState: State{
+			depositRecordAmount: initialDepositAmount,
+			hostZone:            hostZone,
+		},
+		validMsg: stakeibc.MsgRedeemStake{
+			Creator:  user.acc.String(),
+			Amount:   redeemAmount,
+			HostZone: "GAIA",
+			// TODO set this dynamically through test helpers for host zone
+			Receiver: "cosmos1g6qdx6kdhpf000afvvpte7hp0vnpzapuyxp8uf",
+		},
+	}
+}
+
+func (suite *KeeperTestSuite) TestRedeemStakeSuccessful() {
+	tc := suite.SetupRedeemStake()
+
+	msg := tc.validMsg
+	user := tc.user
+	redeemAmount := sdk.NewInt(msg.Amount)
+
+	// get the initial unbonding amount *before* calling liquid stake, so we can use it to calc expected vs actual in diff space
+	epochUnbondingRecordStart, found := suite.App.RecordsKeeper.GetLatestEpochUnbondingRecord(suite.Ctx)
```

Comment:
I think this is a good example where it's actually easier to to just pass back this information from the SetupRedeemStake function! For instance, we're returning the `intialDepositAmount` (which is unnecessary) - we could just as easily return the `initialEpochUnbondingAmount` instead. 

Apologies for misleading you a little here with the earlier comment. But I think when it comes to defining an expected value, we generally want that to be closer to a hard coded value, and avoid checking the state if possible (just to keep transformations minimal).

There's definitely two schools of thought on this. I think the bank supply is another example where I would probably do the opposite and just query the bank before hand, rather than try to keep track of all the balances that were funded. Curious to hear your opinion on it though!

Reply:
Great suggestion, I agree fully. Addressed in current commit